Bluetooth: Mesh: Use updated Model Extension API (#5156)
nrfconnect/sdk-nrf: doc/nrf/libraries/bluetooth_services/mesh/scene_srv.rst

* Scene Server provides functionality for listing and recalling stored scenes.
* Scene Setup Server provides functionality for storing and deleting scenes.

The Scene Setup Server model extends the Scene Server model and the Generic Default Transition Time Server model.

> The Generic Default Transition Time Server model shall be present on the same element or on an element with a lower address
> than the Scene Setup Server is on.

Reviewer: ```suggestion
The Generic Default Transition Time Server model shall be present on the same element as the Scene Setup Server model, or on an element with a lower address.
```
```suggestion
The Generic Default Transition Time Server model shall be present on the same element as the Scene Setup Server model, or on an element with a lower address.
```
